pkg/common/x509util/cert.go

```diff
@@ -43,6 +43,22 @@ func CreateCertificate(ctx context.Context, km keymanager.KeyManager, template,
 	return cert, nil
 }
 
+func DedupeCertificates(bundles ...[]*x509.Certificate) []*x509.Certificate {
+	certMap := map[string]*x509.Certificate{}
+	for _, bundle := range bundles {
+		for _, cert := range bundle {
+			certMap[string(cert.Raw)] = cert
+		}
+	}
+
+	certs := []*x509.Certificate{}
+	for _, cert := range certMap {
```

Comment:
A side effect of iterating over this map to build the result is that the ordering of the returned certificates will not be stable given the same input. That may or may not be important but we can get stable ordering by just appending certs to the output slice as long as they haven't been seen before (e.g. track "seen" status via `map[string]struct{}`).

Reply:
This is important for testing. I fixed it